src/settings/settings.go

```diff
@@ -9,29 +11,33 @@ type Settings struct {
-	StatsdPort                 int    `envconfig:"STATSD_PORT" default:"8125"`
-	RuntimePath                string `envconfig:"RUNTIME_ROOT" default:"/srv/runtime_data/current"`
-	RuntimeSubdirectory        string `envconfig:"RUNTIME_SUBDIRECTORY"`
-	RuntimeIgnoreDotFiles      bool   `envconfig:"RUNTIME_IGNOREDOTFILES" default:"false"`
-	LogLevel                   string `envconfig:"LOG_LEVEL" default:"WARN"`
-	RedisSocketType            string `envconfig:"REDIS_SOCKET_TYPE" default:"unix"`
-	RedisUrl                   string `envconfig:"REDIS_URL" default:"/var/run/nutcracker/ratelimit.sock"`
-	RedisPoolSize              int    `envconfig:"REDIS_POOL_SIZE" default:"10"`
-	RedisAuth                  string `envconfig:"REDIS_AUTH" default:""`
-	RedisTls                   bool   `envconfig:"REDIS_TLS" default:"false"`
-	RedisPerSecond             bool   `envconfig:"REDIS_PERSECOND" default:"false"`
-	RedisPerSecondSocketType   string `envconfig:"REDIS_PERSECOND_SOCKET_TYPE" default:"unix"`
-	RedisPerSecondUrl          string `envconfig:"REDIS_PERSECOND_URL" default:"/var/run/nutcracker/ratelimitpersecond.sock"`
-	RedisPerSecondPoolSize     int    `envconfig:"REDIS_PERSECOND_POOL_SIZE" default:"10"`
-	RedisPerSecondAuth         string `envconfig:"REDIS_PERSECOND_AUTH" default:""`
-	RedisPerSecondTls          bool   `envconfig:"REDIS_PERSECOND_TLS" default:"false"`
-	ExpirationJitterMaxSeconds int64  `envconfig:"EXPIRATION_JITTER_MAX_SECONDS" default:"300"`
-	LocalCacheSizeInBytes      int    `envconfig:"LOCAL_CACHE_SIZE_IN_BYTES" default:"0"`
+	Port                         int           `envconfig:"PORT" default:"8080"`
+	GrpcPort                     int           `envconfig:"GRPC_PORT" default:"8081"`
+	DebugPort                    int           `envconfig:"DEBUG_PORT" default:"6070"`
+	UseStatsd                    bool          `envconfig:"USE_STATSD" default:"true"`
+	StatsdHost                   string        `envconfig:"STATSD_HOST" default:"localhost"`
+	StatsdPort                   int           `envconfig:"STATSD_PORT" default:"8125"`
+	RuntimePath                  string        `envconfig:"RUNTIME_ROOT" default:"/srv/runtime_data/current"`
+	RuntimeSubdirectory          string        `envconfig:"RUNTIME_SUBDIRECTORY"`
+	RuntimeIgnoreDotFiles        bool          `envconfig:"RUNTIME_IGNOREDOTFILES" default:"false"`
+	LogLevel                     string        `envconfig:"LOG_LEVEL" default:"WARN"`
+	RedisSocketType              string        `envconfig:"REDIS_SOCKET_TYPE" default:"unix"`
+	RedisUrl                     string        `envconfig:"REDIS_URL" default:"/var/run/nutcracker/ratelimit.sock"`
+	RedisPoolSize                int           `envconfig:"REDIS_POOL_SIZE" default:"10"`
+	RedisAuth                    string        `envconfig:"REDIS_AUTH" default:""`
+	RedisTls                     bool          `envconfig:"REDIS_TLS" default:"false"`
+	RedisPipelineWindow          time.Duration `envconfig:"REDIS_PIPELINE_WINDOW" default:"75µs"`
```

Comment:
Defaults for pipeline window and limit are chosen according to my e2e benchmark results. Also have a look at [this issue](https://github.com/mediocregopher/radix/issues/147).

I set up a 8c8g envoy with 10 2c2g ratelimit servers, and got 60000 requests/second . `75us` window seems better than `150us`(radix default) in this load.